File: content/search/source/includes/partial-match-tutorial/code-snippets/java/CreateStringIndex.java
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class CreateStringIndex {
    public static void main(String[] args) {
        // connect to your Atlas deployment
        String uri = "<connection-string>";

        try (MongoClient mongoClient = MongoClients.create(uri)) {
            // set namespace
            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("movies");

            // define your MongoDB Search index
            Document indexDefinition = new Document("mappings", 
                new Document("dynamic", false)
                    .append("fields", 
                        new Document("title", 
                            new Document("type", "string")
                        )
                    )
            );

            // run the helper method
            String result = collection.createSearchIndex("partial-match-tutorial", indexDefinition);
            System.out.println("New index name: " + result);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: content/search/source/includes/partial-match-tutorial/code-snippets/python/createStringIndex.py
from pymongo import MongoClient

# connect to your Atlas deployment
uri = "<connection-string>"

client = MongoClient(uri)

def run():
    try:
        # set namespace
        database = client["sample_mflix"]
        collection = database["movies"]

    # define your MongoDB Search index
        index = {
            "name": "partial-match-tutorial",
            "definition": {
                # search index definition fields
                "mappings": {
                    "dynamic": False,
                    "fields": {
                        "title": {
                            "type": "string",
                        }
                    }
                }
            }
        }

        # run the helper method
        result = collection.create_search_index(index)
        print(f"New index name: {result}")
    finally:
        client.close()

if __name__ == "__main__":
    run()
